src/components/Main.tsx

```diff
@@ -0,0 +1,153 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useCallback, useContext, useMemo, useState } from 'react';
+import { Todo } from '../types/Todo';
+import classNames from 'classnames';
+import { ErrorContext, LoadingContext, TodosContext } from './TodoContext';
+import { deleteTodos, editTodos } from '../api/todos';
+import { Status } from '../types/Status';
+import { getFilteredItems } from '../services/getFilteredItems';
+
+type Props = {
+  query: Status;
+};
+
+export const Main: React.FC<Props> = ({ query }) => {
+  const { list, setList } = useContext(TodosContext);
+  const { setErrorMessage } = useContext(ErrorContext);
+  const { isLoading } = useContext(LoadingContext);
+  const [editingTodo, setEditingTodo] = useState<Todo | null>(null);
+  const listToGo = useMemo(() => {
+    return getFilteredItems(list, query);
+  }, [list, query]);
+
+  const handleTodoDelete = useCallback(
+    (todoId: number) => {
+      deleteTodos(todoId).catch(() =>
+        setErrorMessage('Unable to delete a todo'),
+      );
+
+      setList(currentTodos => {
+        return currentTodos.filter(currentTodo => currentTodo.id !== todoId);
+      });
```

Comment:
- You should remove todo from the todo list only if the request was successful 
- And you should show the loader to a user so they know that something happening. You can store todo ids that currently loading in an array, and depending on it show corresponding loaders for todos

The same for other handlers